tests/test_utils.py

```diff
@@ -44,3 +46,33 @@ def test_temporal_splits():
         }
     ]
     assert_split(splits, expected)
+
+
+def test_filename_friendly_hash():
```

Comment:
I think it would be good to have a second test that passes a dict of dicts and see if the hash function is stable. This might happen when the outcome label is actually a lookup table.